### ISSUE TITLE
Know when to be multiline by (1) counting matching brackets (2) a blank newline means force it

### ISSUE THREAD

**Issue by fzyzcjy** [fzyzcjy/dart_interactive#18]

**Comment by fzyzcjy** [comment]
hard to mock, so just do UT instead

**Comment by fzyzcjy** [comment]
```
(base) ➜  interactive git:(master) ✗ dart run --enable-vm-service bin/interactive.dart
The Dart VM service is listening on http://127.0.0.1:8181/Io3T7rCkyy4=/
The Dart DevTools debugger and profiler is available at: http://127.0.0.1:8181/Io3T7rCkyy4=/devtools/#/?uri=ws%3A%2F%2F127.0.0.1%3A8181%2FIo3T7rCkyy4%3D%2Fws
executionWorkspaceMain called and sleep
>>> void f(){}
>>> f(
    )
[WARNING 2022-10-22 18:19:29.416018] Error: Hot reload failed, maybe because code has syntax error?
>>> f(
    );
>>>
